astropy/coordinates/angles/errors.py

```diff
@@ -0,0 +1,164 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Custom angle errors and exceptions used in :mod:`astropy.coordinates.angles`."""
+
+__all__ = []
+
+from astropy.utils.exceptions import AstropyWarning
+
+
+class RangeError(ValueError):
+    """
+    Raised when some part of an angle is out of its valid range.
+    """
+
+
+class BoundsError(RangeError):
+    """
+    Raised when an angle is outside of its user-specified bounds.
+    """
+
+
+class IllegalHourError(RangeError):
+    """
+    Raised when an hour value is not in the range [0,24).
+
+    Parameters
+    ----------
+    hour : int, float
+
+    Examples
+    --------
+    .. code-block:: python
+
+        if not 0 <= hr < 24:
+           raise IllegalHourError(hour)
+    """
+
+    def __init__(self, hour):
+        self.hour = hour
+
+    def __str__(self):
+        return (
```

Comment:
Yikes, a bit worrying that none of these appear to be tested! (Out of scope here, obviously.)

Reply:
Agreed!